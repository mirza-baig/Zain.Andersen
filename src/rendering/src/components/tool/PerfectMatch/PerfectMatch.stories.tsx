// Global
import { Story, Meta } from '@storybook/react';
// Local
import PerfectMatch, { PerfectMatchProps } from './PerfectMatch';
import defaultData from './PerfectMatch.mock-data';

export default {
  title: 'PerfectMatch/PerfectMatch',
  component: PerfectMatch,
} as Meta;
const Template: Story<PerfectMatchProps> = (props) => <PerfectMatch {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
