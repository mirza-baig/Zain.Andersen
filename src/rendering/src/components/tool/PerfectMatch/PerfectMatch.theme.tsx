// Lib
import { ThemeFile, ThemeName } from 'lib/context/ThemeContext';

type PerfectMatchThemeType = {
  [key in ThemeName]: PerfectMatchThemeSubType;
};

export type PerfectMatchThemeSubType = {
  classes: {
    headline: string;
    mainBackground: string;
    mainBackgroundImage: string;
    whiteOverlay: string;
    module: string;
    logo: string;
    logoImage: string;
    app: string;
    panelIntro: string;
    panelType: string;
    leftVisible: string;
    leftHeader: string;
    leftTypeHeader: string;
    panelBackground: string;
    secondary: string;
    tertiary: string;
    letsGo: string;
    letsGoBefore: string;
    small: string;
    orange: string;
    large: string;
    orangeShadow: string;
    rightVisible: string;
    typeBtns: string;
    typeButtons: string;
    headingWrapper: string;
    stepHeading: string;
    rteStepCopy: string;
    contentContainer: string;
    stepContentVisible: string;
    optionContainer: string;
    stepSizeAndPanels: string;
    top: string;
    optionImage: string;
    optionHeading: string;
    optionSizes: string;
    optionSizesDisplayNone: string;
    widthSelect: string;
    heightSelect: string;
    sizeAndPanelsOption: string;
    sizeAndPanelsOptionNoSize: string;
    panelSelected: string;
    widthDropdown: string;
    heightDropdown: string;
    notSureLabel: string;
    notSureCheckbox: string;
    sizeAndPanelsNext: string;
    map: string;
    selectWrapper: string;
    selectClass: string;
    buttonNext: string;
    panelIntroImage: string;
  };
};

export const PerfectMatchTheme = (): ThemeFile | PerfectMatchThemeType => {
  return {
    aw: {
      classes: {
        /** Insert Theme classes here **/
        headline: 'abc-123',
        mainBackground: `main-background h-full w-full fixed top-0 left-0 transition-all -z-20 `,
        mainBackgroundImage: ` h-full w-full object-cover object-center `,
        whiteOverlay:
          'white-overlay fixed top-0 left-0 -z-10 h-full w-full bg-white opacity-75 transition-all transition-opacity',
        module: 'perfect-match-module font-futura-pt ',
        logo: `logo fixed top-0 p-[32px_0_0_40px] cursor-pointer border-none bg-none `,
        logoImage: `  `,
        app: 'app',
        panelIntro: 'panel intro flex flex-col mmd:flex-row pt-20 md:pt-0',
        panelType: 'panel type flex flex-col w-full items-start ',
        leftVisible:
          'left visible w-full relative m-[12px_auto] max-w-[500px] md:m-[40px_0_0_24px] max-w-full p-[20px_10px] ',
        leftHeader: ` max-w-[510px] `,
        leftTypeHeader: ` max-w-[510px] text-[56px] leading-[1.25] mb-[12px] font-bold uppercase `,
        panelBackground: 'block h-[100%] absolute top-0 w-full -z-1',
        secondary: 'secondary text-[48px] leading-[1] font-bold ',
        tertiary: 'tertiary text-[24px] leading-[1] ',
        letsGo:
          'start btn help-me-choose hmc-click inline-block w-full md:w-auto min-h-[44px] p-[20px] md:p-[20px_100px] m-[32px_auto] md:m-[32px_16px] rounded-[2px] bg-white/50 hover:bg-white/75 border-2 border-primary font-bold text-primary relative text-center uppercase text-[18px] leading-[1] ',
        letsGoBefore: ` before:border-t-[rgba(0,0,0,0)] before:border-r-[rgba(0,0,0,0)] before:border-b-primary before:border-l-[rgba(0,0,0,0)] before:border-t-[0] before:border-r-[32px] before:border-b-[32px] before:border-l-[0] before:absolute before:h-0 before:bottom-[2px] before:left-[2px] before:w-0 `,
        small: 'small text-[36px] leading-[1] font-bold uppercase ',
        orange: 'text-primary',
        large: 'large text-[72px] leading-[56px] font-bold uppercase ',
        orangeShadow: ' text-primary drop-shadow-[-1px_-1px_#fff] ',
        rightVisible:
          'right visible flex flex-auto flex-col md:flex-row items-center justify-center p-[20px_10px] mx-auto w-full ',
        typeBtns: 'type-btns hidden md:flex justify-end items-center p-[24px_0_0_0] max-w-[500px] ',
        typeButtons: 'type btn hmc-click text-[18px] leading-[1] min-h-[44px] p-[12px_16px] z-0 ',
        headingWrapper: 'heading-wrapper max-w-[1200px] mx-auto ',
        stepHeading:
          'step-heading font-sans text-black text-[26px] leading-[1.25] md:text-[34px] mb-[12px] font-bold text-center uppercase  ',
        rteStepCopy:
          'step-copy font-sans text-[18px] leading-[1.5] text-black mb-[20px] md:text-[18px] text-center font-light ',
        contentContainer: 'content-container max-w-[1200px] mx-auto ',
        stepContentVisible: 'step-content visible',
        optionContainer:
          'option-container flex flex-wrap justify-center lg:px-[15px] max-w-[1200px] my-0 mx-auto ',
        stepSizeAndPanels: 'step-content step-size-and-panels visible text-center',
        top: 'top flex flex-col h-full ',
        optionImage: 'image p-[60px_20px_10px_20px] flex flex-initial justify-center items-center ',
        optionHeading: 'heading flex-initial p-[10px] text-[14px] md:text-[22px] grow ',
        optionSizes: 'sizes w-full p-[20px]',
        optionSizesDisplayNone: 'sizes sizes-not-visible hidden ',
        widthSelect: 'width-select w-full',
        heightSelect: 'height-select w-full',
        sizeAndPanelsOption:
          'bouncyCard btn option hmc-click is-icon is-preference three-column is-size hover:no-underline md:transition-all m-[10px_5px_10px_5px] md:m-[0_15px_40px_15px] bg-white md:bg-white/50 md:hover:bg-white/75 md:outline-none md:transition-all basis-[44%] md:basis-[30%] md:hover:scale-[1.1] border-gray',
        sizeAndPanelsOptionNoSize:
          'bouncyCard btn option hmc-click is-icon is-preference three-column hover:no-underline md:transition-all m-[10px_5px_10px_5px] md:m-[0_15px_40px_15px] bg-white md:bg-white/50 md:hover:bg-white/75 md:outline-none md:transition-all basis-[44%] md:basis-[30%] md:hover:scale-[1.1] border-gray',
        panelSelected: ` selected border !border-white shadow-[0_0_0_3px] shadow-primary `,
        widthDropdown: 'width-dropdown text-[14px] leading-[1] capitalize text-left ',
        heightDropdown: 'height-dropdown text-[14px] leading-[1] capitalize text-left ',
        notSureLabel: ` flex flex-row text-black text-[14px] uppercase max-w-full mt-[5px] mb-[7px] `,
        notSureCheckbox: 'not-sure-checkbox',
        sizeAndPanelsNext:
          'next btn last-btn primary hmc-click text-white uppercase p-[16px] text-[18px] leading-[18px] mb-[20px] bg-primary border-primary border disabled disabled:opacity-[.65] disabled:pointer-events-none  ',
        map: 'map hidden mmd:block [&_svg]:mx-auto [&_svg]:max-w-full [&_path]:pointer-events-auto [&_path]:transition-all [&_path]:duration-300 [&_path:hover]:cursor-pointer [&_path:hover]:fill-primary',
        selectWrapper: 'col-span-12 mx-auto mt-0 mb-5 block p-[20px_10px] mmd:hidden',
        selectClass: 'mb-5 w-full',
        buttonNext: 'h-11 w-full bg-primary text-white',
        panelIntroImage: ' block h-auto m-[0_auto] max-w-[100%]',
      },
    },
    rba: {
      classes: {
        /** Insert Theme classes here **/
      },
    },
  };
};
