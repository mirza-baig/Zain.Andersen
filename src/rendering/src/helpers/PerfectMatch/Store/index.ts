import { create } from 'zustand';
import { devtools, persist } from 'zustand/middleware';
import { StepSlice, createStepSlice } from './step-slice';
import { SelectionsSlice, createSelectionsSlice } from './selections-slice';
import { TypeSlice, createTypeSlice } from './type-slice';
import { FiltersSlice, createFiltersSlice } from './filters-slice';
import { LocationSlice, createLocationSlice } from './location-slice';

type PerfectMatchState = StepSlice & SelectionsSlice & TypeSlice & FiltersSlice & LocationSlice;
export default PerfectMatchState;

export const usePerfectMatchState = create<PerfectMatchState>()(
  devtools(
    persist(
      (...a) => ({
        ...createStepSlice(...a),
        ...createSelectionsSlice(...a),
        ...createTypeSlice(...a),
        ...createFiltersSlice(...a),
        ...createLocationSlice(...a),
      }),
      { name: 'perfectMatchStore' }
    )
  )
);
