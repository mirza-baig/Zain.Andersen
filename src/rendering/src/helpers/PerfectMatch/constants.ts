import { PerfectMatchSteps } from './types';

export const Steps = {
  Intro: 'intro' as PerfectMatchSteps,
  Type: 'type' as PerfectMatchSteps,
  WindowStyle: 'windowStyle' as PerfectMatchSteps,
  WindowLocation: 'windowLocation' as PerfectMatchSteps,
  WindowCoastal: 'windowCoastal' as PerfectMatchSteps,
  WindowProjectType: 'windowProjectType' as PerfectMatchSteps,
  WindowPreferences: 'windowPreferences' as PerfectMatchSteps,
  WindowResults: 'windowResults' as PerfectMatchSteps,
  DoorStyle: 'doorStyle' as PerfectMatchSteps,
  DoorOperation: 'doorOperation' as PerfectMatchSteps,
  DoorSizeAndPanels: 'doorSizeAndPanels' as PerfectMatchSteps,
  DoorPanelStyle: 'doorPanelStyle' as PerfectMatchSteps,
  DoorLocation: 'doorLocation' as PerfectMatchSteps,
  DoorCoastal: 'doorCoastal' as PerfectMatchSteps,
  DoorProjectType: 'doorProjectType' as PerfectMatchSteps,
  DoorPreferences: 'doorPreferences' as PerfectMatchSteps,
  DoorResults: 'doorResults' as PerfectMatchSteps,
};
