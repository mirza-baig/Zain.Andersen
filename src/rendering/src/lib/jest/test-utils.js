// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
