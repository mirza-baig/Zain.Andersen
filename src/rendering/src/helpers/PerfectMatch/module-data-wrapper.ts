import { URLSearchParams } from 'url';
import { PerfectMatchFilters } from './Store/filters-slice';
import { PerfectMatchModuleData, PerfectMatchProperties, PerfectMatchSteps } from './types';

export class ModuleDataWrapper {
  data: PerfectMatchModuleData;

  constructor(data: PerfectMatchModuleData) {
    this.data = data;
  }

  findValuesForResults(filters: PerfectMatchFilters, propertyName: PerfectMatchProperties) {
    const results = this.findResultsForFilters(filters);
    return results.reduce(
      (found, currentResult) =>
        currentResult.options[propertyName]
          ? found.concat(currentResult.options[propertyName] ?? [])
          : found,
      [] as string[]
    );
  }

  findOptionsForResults(filters: PerfectMatchFilters, propertyName: PerfectMatchProperties) {
    // Build the list of all available options
    const matchingOptionIds = this.findValuesForResults(filters, propertyName);

    // We could have duplicates, so filter them out
    const options = this.data.options.filter((option) => matchingOptionIds.indexOf(option.id) > -1);

    // Sort the options
    options.sort((a, b) => a.sortOrder - b.sortOrder);

    return options;
  }

  getUniqueWidthsAndHeightsForResultsAndOption(filters: PerfectMatchFilters, optionId: string) {
    const widths = [] as string[];
    const heights = [] as string[];

    const sizes = this.findValuesForResults(
      filters,
      ('productSize:' + optionId) as PerfectMatchProperties
    );

    if (sizes.length > 0) {
      sizes.forEach((size) => {
        const parts = size.split('x');

        if (widths.indexOf(parts[0]) === -1) {
          widths.push(parts[0]);
        }

        if (heights.indexOf(parts[1]) === -1) {
          heights.push(parts[1]);
        }
      });

      widths.sort();
      heights.sort();
    }

    return { widths, heights };
  }

  findResultsForFilters(filters: PerfectMatchFilters) {
    let results = this.data.results;

    for (const property in filters) {
      results = results.filter(function (result) {
        const options = result.options[property]?.filter(function (id) {
          const filter = filters[property as PerfectMatchProperties];
          return filter && filter.indexOf(id) > -1;
        });
        return options && options.length > 0;
      });
    }
    return results;
  }

  getIsWindow(typeId: string) {
    return typeId === this.data.optionIds.type.window;
  }
  getIsPatioDoor(typeId: string) {
    return typeId === this.data.optionIds.type.patioDoor;
  }

  getCurrentStepObject(stepName: PerfectMatchSteps | string) {
    return this.data.steps[stepName];
  }
}

export const SetFilters = (parsedHash: URLSearchParams, moduleData: PerfectMatchModuleData) => {
  const productType = parsedHash.get('s:type') as string;
  const doorStyle = parsedHash.get('s:doorStyle') as string;
  const windowStyle = parsedHash.get('s:windowStyle') as string;
  if (productType) {
    moduleData.filters.productType = productType;
  }
  if (doorStyle) {
    moduleData.filters.productStyle = doorStyle;
  }
  if (windowStyle) {
    moduleData.filters.productStyle = windowStyle;
  }
};
