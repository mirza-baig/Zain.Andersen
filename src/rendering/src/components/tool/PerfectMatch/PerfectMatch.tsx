// Global
import { Image, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { useTheme } from 'lib/context/ThemeContext';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
//import { Component } from 'src/helpers/Component';
import { useRouter } from 'next/router';
import { createContext, useContext, useEffect, useState } from 'react';
import { ModuleDataWrapper } from 'src/helpers/PerfectMatch/module-data-wrapper';
import { SetFilters } from 'src/helpers/PerfectMatch/perfect-match-mappers';
import {
  MapOptionIds,
  MapOptions,
  MapPanelOptionHingedIcons,
  MapPreferences,
  MapProjectTypes,
  MapRegions,
  MapResults,
  MapSteps,
} from 'src/helpers/PerfectMatch/perfect-match-mappers';
import RenderSteps from 'src/helpers/PerfectMatch/Steps/PerfectMatchStep.helper';
import Steps from 'src/helpers/PerfectMatch/Steps/Steps.helper';
import { usePerfectMatchState } from 'src/helpers/PerfectMatch/Store';
import { PerfectMatchModuleData, PerfectMatchSteps } from 'src/helpers/PerfectMatch/types';
import { PerfectMatchTheme, PerfectMatchThemeSubType } from './PerfectMatch.theme';

export type PerfectMatchProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.Tool.PerfectMatch.PerfectMatch;

export type PerfectMatchOptionProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.Tool.PerfectMatch.PerfectMatchOption;
export type PerfectMatchResultProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.Tool.PerfectMatch.PerfectMatchResult;

export const PerfectMatchDataContext = createContext<ModuleDataWrapper>({} as ModuleDataWrapper);

export const usePerfectMatchDataContext = () => useContext(PerfectMatchDataContext);

const PerfectMatch = (props: PerfectMatchProps) => {
  const { themeData } = useTheme(PerfectMatchTheme());
  const theme = (themeData as PerfectMatchThemeSubType).classes;
  const [navigationVisible, setNavigationVisible] = useState<boolean>(false);

  const moduleData: PerfectMatchModuleData = {
    steps: MapSteps(props),
    results: MapResults(props),
    preferences: MapPreferences(props),
    projectTypes: MapProjectTypes(props),
    regions: MapRegions(props),
    optionIds: MapOptionIds(props),
    options: MapOptions(props),
    panelOptionHingedIcons: MapPanelOptionHingedIcons(props),
    filters: {},
  };

  /* eslint-disable @typescript-eslint/no-explicit-any */
  const currentStep = usePerfectMatchState((x: any) => x.currentStep);
  const setCurrentStep = usePerfectMatchState((x: any) => x.setCurrentStep);
  /* eslint-enable @typescript-eslint/no-explicit-any */

  const currentStepNumber = moduleData.steps[currentStep].stepNumber ?? 0;
  const maxStepNumber = Math.max(
    ...Object.keys(moduleData.steps).map((key: string) => {
      return moduleData.steps[key]?.stepNumber ?? 0;
    })
  );

  const router = useRouter();

  const moduleWrapper = new ModuleDataWrapper(moduleData);

  useEffect(() => {
    if (router.isReady) {
      const hash = (router.asPath as string).split('#')[1];
      const parsedHash = new URLSearchParams(hash);

      const step = parsedHash.get('cs') as PerfectMatchSteps;

      SetFilters(parsedHash, moduleData);

      if (moduleData?.steps?.hasOwnProperty(step)) {
        setCurrentStep(step);
      }
    }
  }, [moduleData.steps, router.isReady, router.asPath, setCurrentStep]);

  const RenderCSSHidingBlock = () => {
    return (
      <>
        <style>
          {`
          @media screen and (min-width:672px)
          {
            header #header, section.breadcrumbs-module, footer #footer
            {
              visibility: hidden;
              opacity: 0;
              transition: visibility 0s linear 300ms, opacity 300ms;
            }

            footer #footer
            {
              display: none;
            }
          }

          @media print {
            #header, #footer {
              display: none;
            }
          }`}
        </style>
      </>
    );
  };

  return (
    <PerfectMatchDataContext.Provider value={moduleWrapper}>
      <section className={theme.module} data-component="perfectmatch/perfectmatch">
        {router.isReady && currentStep && currentStepNumber < maxStepNumber ? (
          <div className={theme.mainBackground}>
            <Image
              field={props.fields?.moduleBackgroundImage}
              className={theme.mainBackgroundImage}
            ></Image>
          </div>
        ) : (
          <></>
        )}
        {router.isReady &&
        currentStep &&
        currentStepNumber < maxStepNumber &&
        currentStepNumber > 1 ? (
          <div className={theme.whiteOverlay}></div>
        ) : (
          <></>
        )}
        {!navigationVisible && currentStepNumber < maxStepNumber && (
          <button
            className={theme.logo}
            title="Show Navigation"
            aria-label="Show Navigation"
            onClick={() => setNavigationVisible(true)}
          >
            <Image field={props.fields?.logoImage} className={theme.logoImage}></Image>
          </button>
        )}
        <div className={theme.app}>
          {currentStepNumber - 2 >= 0 && currentStepNumber < maxStepNumber ? (
            <Steps props={props} step={currentStep} moduleData={moduleData}></Steps>
          ) : (
            <></>
          )}
          <RenderSteps props={props} step={currentStep} moduleData={moduleData}></RenderSteps>
        </div>
      </section>
      {!navigationVisible && currentStepNumber < maxStepNumber && RenderCSSHidingBlock()}
    </PerfectMatchDataContext.Provider>
  );
};

export default withDatasourceCheck()<PerfectMatchProps>(PerfectMatch);
