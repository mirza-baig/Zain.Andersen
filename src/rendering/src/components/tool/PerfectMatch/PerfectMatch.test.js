// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import PerfectMatch from './PerfectMatch';
import defaultData from './PerfectMatch.mock-data';

jest.mock('next/router', () => ({
  useRouter() {
    return ({
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn()
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null)
    });
  },
}));

it('renders correctly', async () => {
  const component = snapshot(PerfectMatch, { componentProps: defaultData });
  await hasDataComponent(component, 'perfectmatch/perfectmatch');
});
