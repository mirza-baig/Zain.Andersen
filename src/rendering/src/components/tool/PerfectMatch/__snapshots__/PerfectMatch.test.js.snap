// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  class="perfect-match-module font-futura-pt "
  data-component="perfectmatch/perfectmatch"
>
  <button
    aria-label="Show Navigation"
    class="logo fixed top-0 p-[32px_0_0_40px] cursor-pointer border-none bg-none "
    title="Show Navigation"
  >
    <img
      alt="Andersen logo with hamburger menu for Perfect Match tool"
      class="  "
      height="23"
      src="https://preview.local.andersenwindows.com/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Perfect-Match/perfect-match-logo.png?h=23&iar=0&w=200&rev=a3a5108c7fa843b3b2048a19c360d3a6&hash=351287C9DD33DEC859840B20CE9AD0D2"
      width="200"
    />
  </button>
  <div
    class="app"
  />
</section>
`;
